src/frida/repl.py

```diff
@@ -51,6 +56,13 @@ def _start(self):
                 self._update_status("Failed to load script: {error}".format(error=e))
                 self._exit(1)
                 return
+
+            if self._user_script is not None and self._watch:
+                monitor = frida.FileMonitor(self._user_script)
+                monitor.on('change', self._on_change)
+                monitor.enable()
+                self._script_monitor = monitor
+
```

Comment:
This will not work if the user uses the `%load` magic command from the REPL. The solution is simple though: if we extract out the call to `self._create_repl_script()` to the line before, having it assign to a `source` variable, we should do the monitoring right after this line. Similarly by moving the teardown to `_unload_script()`, which gets called by `_stop()`, we will have teardown working correctly also.